docs/TektonConfig.md

```diff
@@ -408,3 +426,84 @@ platforms:
 [priorityClassName]: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority
 [priorityClass]: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass
 
+### Additional fields as `options`
+There is a filed called `options` available in all the components.<br>
+
+>**NOTE:** There is a possibility to have two different values for a field.<br> 
+An example: with a pre-defined field you can set value and the same filed may be defined under `options` as well. In that case value from `options` will be final.
+
+`options` filed is defined as follows,
+```yaml
+options:
+  disabled: false
+  configMaps:
+    config-leader-election:
+      metadata:
+        labels:
+        annotations:
+      data:
+        buckets: "2"
+    my-custom-config-map: # creates new configMap under targetNamespace
+      data:
+        foo: bar
+  deployments:
+    tekton-pipelines-controller:
+      metadata:
+        labels:
+          custom-label: "foo"
+        annotations:
+          custom-annotation: "foo"
+      spec:
+        replicas: 2
+        template:
+          spec:
+            affinity:
+            nodeSelector:
+            tolerations:
+            topologySpreadConstraints:
+            volumes:
+              - name: config-logging
+                configMap:
+                  name: pipeline-config-logging
+            initContainers:
+            containers:
+              - name: tekton-pipelines-controller
+                volumeMounts:
+                env:
+                  - name: CONFIG_LOGGING_NAME
+                    value: pipeline-config-logging
+                args:
+                  - "--disable-ha=false"            
+```
+* `disabled` - disables the additional `options` support, if `disabled` set as `true`. default: `false`
```

Comment:
Any reason to have that `disabled` field, can't we just consider it disabled when there is nothing, and enabled when there is ?

Reply:
@vdemeester at this time this is an experimental feature. Hence I added an option to disable this.

Reply:
and another reason to keep `disabled` field, I want to persist my options, even it is disabled :)

Reply:
Alright 👌🏼